service/collector/env.go

```diff
@@ -76,10 +82,14 @@ func (c *Env) Collect(ch chan<- prometheus.Metric) error {
 		for _, i := range l.Items {
 			desiredClusterIDs = append(desiredClusterIDs, i.Name)
 		}
+
+		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d desired cluster ids", len(desiredClusterIDs)))
```

Comment:
```suggestion
		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d cluster IDs: %#v", len(desiredClusterIDs), desiredClusterIDs))
```
It might make sense to also log the found cluster IDs. Based on the log message, my first reaction would probably be "What did those cluster IDs include? Is there anything missing?"